cfgrib/dataset.py

```diff
@@ -563,10 +568,14 @@ def build_variable_components(
         coord_vars["valid_time"] = Variable(dimensions=time_dims, data=time_data, attributes=attrs)
 
     for coord_name in extra_coords:
-        coord_vars[coord_name] = Variable(
-            dimensions=(extra_coords[coord_name],),
-            data=np.array(list(extra_coords_data[coord_name].values())),
-        )
+        coord_data = np.array(list(extra_coords_data[coord_name].values()))
```

Comment:
Why do you need `np.array(list(`? Am I missing something?

```suggestion
        coord_data = np.array(extra_coords_data[coord_name].values())
```

Reply:
unfortunately yes, I need to apply the list before. Without list, I obtain an array containing only one object dict_values(...) :(